### PR TITLE
Update AnchorSample.cs parent object for anchor instantiation

### PR DESCRIPTION
**Pull request by KBonds**
⭐ **Feel free to do whatever with this branch while I'm out.**

Targeting 1.4.1-again branch for a smaller diff; I'm expecting these two to be merged at the same time.

- Removes the old anchor container from the ARAnchor scene, using ARSessionOrigin.trackablesParent instead
- Removes the logic for processing existing anchors - these are only relevant in this app if there are already anchors in the subsystem upon loading a scene, which would mean a bug had previously occurred in this sample.

Fixes [#1217](https://github.com/microsoft/Unity-MR-Plugin/issues/1217)

### CODE REVIEW

File: BasicSample/Assets/ARAnchor/Scripts/AnchorsSample.cs [hunk 1]
```diff
@@ -73,6 +65,10 @@ protected async void OnEnable()
                 TrackableId trackableId = m_anchorStore.LoadAnchor(name);
                 m_incomingPersistedAnchors.Add(trackableId, name);
             }
+
+            // Anchors may already exist in the subsystem, but are not guaranteed to be loaded into the scene until
+            // the AnchorsChanged event. Setting this flag tells this app to process existing anchors at that time.
+            m_processExistingAnchorsOnAnchorsChanged = true;
```
Comment (satyapoojasama) [inline BasicSample/Assets/ARAnchor/Scripts/AnchorsSample.cs:71]: I dont think we need to set a flag in OnEnable(). Is it not given that AnchorsChanged() gets called only after OnEnable()? Instead, we can have something like "m_processedExistingAnchors" be set once the anchors are processed on the first AnchorsChanged() event.

File: BasicSample/Assets/ARAnchor/Scripts/AnchorsSample.cs [hunk 2]
```diff
@@ -17,11 +17,9 @@ namespace Microsoft.MixedReality.OpenXR.BasicSample
     /// again near these anchors toggles their persistence, backed by the <c>XRAnchorStore</c>.
     /// </summary>
     [RequireComponent(typeof(ARAnchorManager))]
+    [RequireComponent(typeof(ARSessionOrigin))]
     public class AnchorsSample : MonoBehaviour
     {
-        [SerializeField]
-        private GameObject m_anchorsContainer;
```
Comment (satyapoojasama) [inline BasicSample/Assets/ARAnchor/Scripts/AnchorsSample.cs:23]: why is this removed? what is its significance?

File: BasicSample/Assets/ARAnchor/Scripts/AnchorsSample.cs [hunk 3]
```diff
@@ -207,7 +213,11 @@ public void AddAnchor(Pose pose)
         {
 #if AR_FOUNDATION_4_1_1_OR_LATER
             Debug.Log($"Instantiating new GameObject containing an ARAnchor");
-            Instantiate(m_arAnchorManager.anchorPrefab, pose.position, pose.rotation);
+
+            // When instantiating a trackable gameobject, it should be a child of the ARSessionOrigin.trackablesParent GameObject.
+            // If the GameObject is not a child of this trackablesParent, it may not be cleaned up properly in some scenarios,
+            // such as scene changes. This is especially important for applications composed of additive scenes, like this one.
+            Instantiate(m_arAnchorManager.anchorPrefab, pose.position, pose.rotation, m_arSessionOrigin.trackablesParent);
```
Comment (satyapoojasama) [inline BasicSample/Assets/ARAnchor/Scripts/AnchorsSample.cs:220]: In the Main Test project, we stopped using this instantiate method as it is buggy. Do we want to keep using this in the sample project? 
https://github.com/microsoft/Unity-MR-Plugin/blob/main/TestProjects/Main/Assets/Scripts/AnchorsSample.cs#L175

File: BasicSample/Assets/ARAnchor/Scripts/AnchorsSample.cs (hunk 1 above)
Comment (KBonds) [reply BasicSample/Assets/ARAnchor/Scripts/AnchorsSample.cs:71]: I think the real question is whether or not we care to support the scenarios around this being enabled / disabled, as a more robust sample, even if we're not using it that way.

This script could definitely be simplified if we make assumptions like what you suggested, but I've been leaning towards making this a more robust example of best practices, for general use

Comment (KBonds) [reply BasicSample/Assets/ARAnchor/Scripts/AnchorsSample.cs:71]: To support the flag like you mentioned, we'd still need to reset the flag on disable or enable to handle the case where this is enabled -> disabled -> enabled

File: BasicSample/Assets/ARAnchor/Scripts/AnchorsSample.cs (hunk 2 above)
Comment (KBonds) [reply BasicSample/Assets/ARAnchor/Scripts/AnchorsSample.cs:23]: Now we use ARSessionOrigin.trackablesParent now instead, so there's no need for us to have our own object to put anchors in

File: BasicSample/Assets/ARAnchor/Scripts/AnchorsSample.cs (hunk 3 above)
Comment (satyapoojasama) [reply BasicSample/Assets/ARAnchor/Scripts/AnchorsSample.cs:220]: fair enough. The issue we opened with unity is still under discussion. We can change accordingly later.

**Review by satyapoojasama — APPROVED** [review]

**Review by yl-msft — APPROVED** [review]
